Code/Geometry/Trajectory.h

```diff
@@ -0,0 +1,143 @@
+        numPoints and dimension must match the Trajectory which is going to
+        contain this Snapshot
+        \param energy is the energy associated with this set of coordinates
+     */
+    Snapshot(boost::shared_array<double> pos, double energy = 0.0);
+    /*! \return a const pointer to the parent Trajectory
+     */
+    const Trajectory *trajectory() const {
+      return d_trajectory;
+    }
+    /*! \param pointNum is the atom number whose coordinates will be retrieved
+        \return the coordinates at pointNum as a Point2D object;
+        requires the Trajectory dimension to be == 2
+     */
+    Point2D getPoint2D(unsigned int pointNum) const;
+    /*! \param pointNum is the atom number whose coordinates will be retrieved
+        \return the coordinates at pointNum as a Point3D object;
+        requires the Trajectory dimension to be >= 2
+     */
+    Point3D getPoint3D(unsigned int pointNum) const;
+    /*! \return the energy for this Snapshot
+     */
+    double getEnergy() const {
+      return d_energy;
+    };
+    /*! \brief Sets the energy for this Snapshot
+        \param energy the energy value assigned to this Snapshot
+     */
+    void setEnergy(double energy) {
+      d_energy = energy;
+    }
+    /*! \brief Frees the pointer to the array of doubles where the
+        coordinates for this Snapshot are stored
+     */
+  private:
+    // Pointer to the parent Trajectory object
+    const Trajectory *d_trajectory;
+    // Energy for this set of coordinates
+    double d_energy;
+    // shared array to Snapshot coordinates
+    boost::shared_array<double> d_pos;
+};
+
+class Trajectory {
+  public:
+    typedef std::vector<boost::shared_ptr<Snapshot> > SnapshotSPtrVect;
+    /*! \brief Constructor
+        \param dimension represents the dimensionality of this Trajectory's coordinate tuples;
+        this is normally 2 (2D coordinates) or 3 (3D coordinates)
+        \param numPoints is the number of coordinate tuples associated to each Snapshot
+     */
+    Trajectory(unsigned int dimension, unsigned int numPoints);
+    /*! \brief Copy constructor
+     */
+    Trajectory(const Trajectory &other);
+    /*! \return the dimensionality of this Trajectory's coordinate tuples
+     */
+    unsigned int dimension() const {
+      return d_dimension;
+    }
+    /*! \return the number of coordinate tuples associated to each Snapshot
+     */
+    unsigned int numPoints() const {
+      return d_numPoints;
+    }
+    /*! \return the number of Snapshots associated to this Trajectory
+     */
+    size_t size() const {
+      return d_snapshotVect.size();
+    }
+    /*! \brief Appends a Snapshot to this Trajectory
+        \param s is the Snapshot to be added; the Trajectory
+               takes ownership of the snapshot coordinates
+        \return the zero-based index position of the added Snapshot
+     */
+    unsigned int addSnapshot(Snapshot *s);
```

Comment:
I don't think there is a reason to use a pointer.  A Snapshot is a thin wrapper around a const pointer, a double and a boost shared array.  This API could easily be:

```
unsigned int addSnapshot(const Snapshot &s) {
  vector_snapshot.push_back(s);
}
```
And the copy constructor can do all the work.  This completely removes all concept of ownership.